gobblin-config-management/gobblin-config-core/src/main/java/gobblin/config/client/api/ConfigStoreFactoryDoesNotExistsException.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) 2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.config.client.api;
+
+/**
+ * ConfigStoreFactoryDoesNotExistsException indicate the requested {@link ConfigStoreFactory} can not
+ * be found
+ * @author mitu
+ *
+ */
+public class ConfigStoreFactoryDoesNotExistsException extends Exception {
+  private static final long serialVersionUID = -131707505927389860L;
+  
+  private static final String MESSAGE_FORMAT =
+      "Failed to find the config store factory with scheme: %s, because of: %s ";
+  
+  private final String scheme;
+
+  public ConfigStoreFactoryDoesNotExistsException(String scheme, String message) {
+    super(String.format(MESSAGE_FORMAT, scheme, message));
+    this.scheme = scheme;
+  }
+  
+  public ConfigStoreFactoryDoesNotExistsException(String scheme, Exception e) {
+    super(String.format(MESSAGE_FORMAT, scheme, e.getMessage()), e);
```

Comment:
e.toString() tends to give more meaningful messages since it also includes the Exception class.